### PR TITLE
Fixing worldscale for controllers

### LINKED ISSUE #103: The controller positions do not handle world_scale correctly
If you set the ARVROrigin world_scale to 2, then the hand controllers are up in the sky and move past you at twice the speed you walk.  

It looks like their positions need to be scaled down towards the centre of the tracking space by a factor or 2=world_scale.

This parameter works correctly on the Vive -- the hand controllers are where they should be, while the world is smaller.  In my demo was scaling by a factor of 4 in order to have [smaller voxels](https://github.com/Zylann/godot_voxel/issues/105#issuecomment-625317294) in the voxel terrain library.

### PR DESCRIPTION
**Pull request by BastiaanOlij**
Changing scale of controllers based on worldscale and removing the double applying of worldscale to the tracking.

Haven't looked at hand tracking yet and still downloading SDK so I can test (had to reinstall my PC few weeks back, bah)

fixes #103

### CODE REVIEW

**Comment by NeoSpark314** [comment]
Just tried it. Position fix seems to work as intended. But the controller scale is probably Inverse. A world scale of 4.0 makes the player 4.0 larger...:

image

**Comment by NeoSpark314** [comment]
ups; posted in the wrong issue before; with the inverse scale size is correct; but it breaks the current setup of getting the right controller mesh by having a negative scale for the right controller:

image

File: demo/addons/godot_ovrmobile/example_scenes/ARVROriginWithInitAndMovement.gd [hunk 1]
```diff
@@ -234,3 +237,11 @@ func _on_RightTouchController_button_release(button):
 	if (ovr_utilities):
 		# reset the color to neutral again
 		ovr_utilities.set_default_layer_color_scale(Color(1.0, 1.0, 1.0, 1.0));
+
+func _check_worldscale():
+	if was_world_scale != world_scale:
+		was_world_scale = world_scale
+		var inv_world_scale = 1.0 / was_world_scale
```
Comment (NeoSpark314) [inline demo/addons/godot_ovrmobile/example_scenes/ARVROriginWithInitAndMovement.gd:244]: as posted in the comment section earlier; this does not need to be inverse as the "world scale" seems to make the player larger.

File: demo/addons/godot_ovrmobile/example_scenes/ARVROriginWithInitAndMovement.gd [hunk 2]
```diff
@@ -234,3 +237,11 @@ func _on_RightTouchController_button_release(button):
...
+		var controller_scale = Vector3(inv_world_scale, inv_world_scale, inv_world_scale)
+		$"LeftTouchController/left-controller".scale = controller_scale
+		$"RightTouchController/right-controller".scale = controller_scale
```
Comment (NeoSpark314) [inline demo/addons/godot_ovrmobile/example_scenes/ARVROriginWithInitAndMovement.gd:247]: I believe here we would need a `-controller_scale` as the right controller is just a mirrored mesh of the left controller in the demo setup

**Comment by m4gr3d** [comment]
@BastiaanOlij Any updates on this PR? 
We'll release version 2.0 of the plugin to match Godot v3.2.2 release, so it'd be nice to have this fix included as part of the release.

File: demo/addons/godot_ovrmobile/example_scenes/ARVROriginWithInitAndMovement.gd (hunk 2 above)
Comment (BastiaanOlij) [reply demo/addons/godot_ovrmobile/example_scenes/ARVROriginWithInitAndMovement.gd:247]: Hmmm, my gut feeling says that minusing the scale isn't a very nice solution, then again this is a script very specific to this setup so I'll make that change.

**Pull request merged**